[T7][T09-B4]

loveandbejoyful opened the pull request.

okkhoy commented on README.md:
```diff
@@ -1,6 +1,6 @@
 # Address Book (Level 4)
 
-[![Build Status](https://travis-ci.org/se-edu/addressbook-level4.svg?branch=master)](https://travis-ci.org/se-edu/addressbook-level4)
+[![Build Status](https://travis-ci.org/CS2103JAN2017-T09-B4/main.svg?branch=master)](https://travis-ci.org/CS2103JAN2017-T09-B4/main.svg?branch=master)
```
need to update the other links (codacy, coveralls) to reflect your project not addressbook.

okkhoy commented on docs/DeveloperGuide.md:
```diff
@@ -346,74 +346,269 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
-`* * *` | user | find a person by name | locate details of persons without having to go through the entire list
-`* *` | user | hide [private contact details](#private-contact-detail) by default | minimize chance of someone else seeing them by accident
-`*` | user with many persons in the address book | sort persons by name | locate a person easily
+`* * *` | user | obtain a list of tasks due today / this week | plan my time to complete these urgent tasks before they are due
```
how about add/delete tasks?

okkhoy commented on docs/DeveloperGuide.md:
```diff
@@ -346,74 +346,269 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
+
+2a. No command contains letter or sequence of letters typed by user
+
+> Use case ends
```
may be you should indicate to the user no such command?

okkhoy commented on docs/DeveloperGuide.md:
```diff
@@ -346,74 +346,269 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
+
+**MSS**
+
+1. User clicks on "Change Save Location"
```
please remember that, there should be a command equivalent to this functionality

okkhoy commented on docs/DeveloperGuide.md:
```diff
@@ -346,74 +346,269 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
+2. System displays a directory chooser
+3. User selects a directory
+4. System changed the save location to the one selected
+5. System displays a message 
```
can combine 4 and 5 above, else it doesn't follow the format of the use case.

okkhoy commented on docs/DeveloperGuide.md:
```diff
@@ -346,74 +346,269 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
+**MSS**
+
+1. User presses on predefined hotkey
+2. System displays main window
```
this use case seems incomplete. 
may be you need to specify what hot keys do what and then write steps for at least a few of them.

okkhoy commented on docs/DeveloperGuide.md:
```diff
@@ -346,74 +346,269 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
+**MSS**
+
+1. User enters undo command
+2. System undoes last change
```
nitpick: may be "reverts" is a better word than "undoes"?

okkhoy commented on docs/DeveloperGuide.md:
```diff
@@ -346,74 +346,269 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
+2. Should process and respond to user commands within 1s. 
+3. Should conform to the Java coding standard.
+4. Should automatically back up data once a week. 
+5. Should detect any intrusion within 10 seconds.
```
while this is ok, 10 seconds is too long to mess up the system :-P

okkhoy commented on docs/UserGuide.md:
```diff
@@ -16,171 +16,146 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
-1. Download the latest `addressbook.jar` from the [releases](../../../releases) tab.
-2. Copy the file to the folder you want to use as the home folder for your Address Book.
+1. Download the latest `taskmanager.jar` from the [releases](../../../releases) tab.
+2. Copy the file to the folder you want to use as the home folder for your Task Manager.
```
need to add a mock up of the UI that you plan to build now. 
later you can change it to the actual product screenshot.

okkhoy commented on docs/UserGuide.md:
```diff
@@ -16,171 +16,146 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+Formats:
+> * `add <task>`
+> * `add <task>; <duration>`
+> * `add <task>; <due date and time>`
```
can we change the order of the parameters? if so please specify.
if not, think how to improve usability on this aspect!

okkhoy commented on docs/UserGuide.md:
```diff
@@ -16,171 +16,146 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
-> The index refers to the index number shown in the most recent listing.<br>
-> The index **must be a positive integer** 1, 2, 3, ...
+Formats:
+> * `update <parameter> <new_value>` (when task to be updated has already been selected)
```
update which task? how will the user specify which task to update?
how will the user know which task is being updated?

okkhoy commented on docs/UserGuide.md:
```diff
@@ -16,171 +16,146 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
-* **Select** : `select INDEX` <br>
-  e.g.`select 2`
+* **Select** : `select <task>` <br>
+  e.g.`select presentation`
```
can consider providing command summary as a table.

okkhoy reviewed (commented):
a good start to the user guide, folks!
it needs a bit more thinking through and polishing. i am sure, you guys will do the necessary changes before the final submission.

- please ensure readme file reflects your project and not addressbook.
- please update the about us page to reflect your team.

okkhoy commented:
@loveandbejoyful @ang-zy @Crsytral 
Some comments added. Please ack & close the PR after reading comments.

Minor issues:
- Jian Sin, there is something wrong with the way you do the commits? in this PR your commits are not identified as your contributions. please take care :-)

BrandonTJS committed:
Update collated files

BrandonTJS committed:
Merge pull request #240 from CS2103JAN2017-T09-B4/V0.5rc_Brandon

Update collated files

BrandonTJS committed:
Merge pull request #237 from CS2103JAN2017-T09-B4/V0.5rc

V0.5rc Master Merge

Crsytral committed:
Fix list today does not list events today #241

Fixes #241

loveandbejoyful committed:
Add tests for list floating & timed + next & prev

loveandbejoyful committed:
Make tests pass

Crsytral committed:
Add Unit Test for DateTime functions

Crsytral committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into V0.5

Crsytral committed:
Fix Checkstyle Violation

Crsytral committed:
Add Method Comments

Crsytral committed:
Fix Checkstyle Violations

loveandbejoyful committed:
Improve code quality

loveandbejoyful committed:
Merge branch 'V0.5' of https://github.com/CS2103JAN2017-T09-B4/main into V0.5

loveandbejoyful committed:
Fix JUnit tests

loveandbejoyful committed:
Lighten completed event colour in calendar

BrandonTJS committed:
Renamed constants

BrandonTJS committed:
Fixed a constant not renamed properly

BrandonTJS committed:
Editing of Floating Task to Non-Floating with only start date will now default to end date = start date

BrandonTJS committed:
Implemented Comparable and Copy Constructor into DateTime class

BrandonTJS committed:
Edit Commands now checks date range before editing task (Start date cannot be before end date)

BrandonTJS committed:
Fixed typo in message

BrandonTJS committed:
Added TagQualifier and support for finding tags (including margin of error)

BrandonTJS committed:
Merge pull request #255 from CS2103JAN2017-T09-B4/v0.5_Brandon

Fix and Resolves multiple issues #254 #243 #242 #153

Crsytral committed:
Update AboutUs.md

BrandonTJS committed:
Renamed static variables to better match convention

Crsytral committed:
Update List Cmd Unit Test

BrandonTJS committed:
Removed unused constants

BrandonTJS committed:
Code cleanup + resolve issue #202

BrandonTJS committed:
Merge pull request #256 from CS2103JAN2017-T09-B4/v0.5_Brandon

Resolve #202

BrandonTJS committed:
Resolves calendar switching view when commands are executed (issue: #248)

BrandonTJS committed:
Merge pull request #257 from CS2103JAN2017-T09-B4/v0.5_Brandon

Resolves calendar switching view when commands are executed

BrandonTJS committed:
Temporary commented out some methods that no longer have a purpose, requires additional stability testing before actual deletion

Crsytral committed:
Update List Cmd Unit Test

Crsytral committed:
Merge branch 'V0.5' of https://github.com/CS2103JAN2017-T09-B4/main into V0.5

ang-zy committed:
Update AddCommand to support recurring tasks

ang-zy committed:
Move CLI Identifers to CliSyntax class

BrandonTJS committed:
Some renaming to reduce ambiguity

BrandonTJS committed:
New Edit NLP

BrandonTJS committed:
Code cleanup

BrandonTJS committed:
Merge pull request #259 from CS2103JAN2017-T09-B4/v0.5_Brandon

Revamped Edit NLP

BrandonTJS committed:
Extracted Recur variables and methods out from task to make the code neater

ang-zy committed:
Refactor parameters in CliSyntax to arrays

ang-zy committed:
Create utility method to check if a string is in an array

ang-zy committed:
Update EditCommandParser to match refactored code

Crsytral committed:
Fix Unit Test Failing

ang-zy committed:
Merge branch 'V0.5' into Refactor-CliSyntax

Crsytral committed:
Update Save and Load Cmd Unit Test

ang-zy committed:
Merge pull request #260 from CS2103JAN2017-T09-B4/Refactor-CliSyntax

Refactor CliSyntax+edit cmd parser

Crsytral committed:
Remove Lucky Draw Notification

Crsytral committed:
Add Messages Unit Test

ang-zy committed:
Update isFoundIn to accept any number of string arrays

loveandbejoyful committed:
Calendar

ang-zy committed:
Fix compilation error for isValidParameter

loveandbejoyful committed:
Merge branch 'V0.5' of https://github.com/CS2103JAN2017-T09-B4/main into V0.5

ang-zy committed:
Remove broken and unused imports

loveandbejoyful committed:
Merge branch 'V0.5' of https://github.com/CS2103JAN2017-T09-B4/main into V0.5

loveandbejoyful committed:
Resolve JUnit

BrandonTJS committed:
Extracted RecurState out of Task class for less smelly code

BrandonTJS committed:
Fix issue #261

loveandbejoyful committed:
Update UI screenshots

loveandbejoyful committed:
Checkstyle

loveandbejoyful committed:
Delete all day slot

loveandbejoyful committed:
Fix exception when no commands are stored

BrandonTJS committed:
Fixed some JUnit

BrandonTJS committed:
Fix Junit

BrandonTJS committed:
Fixed a test dealing with loading from storage

BrandonTJS committed:
Typo

BrandonTJS committed:
Merge branch 'V0.5' into v0.5_Brandon

BrandonTJS committed:
Merge pull request #262 from CS2103JAN2017-T09-B4/v0.5_Brandon

Cleaned up Recurring Task implementation + JUnit Fixes

BrandonTJS committed:
JUnit Fix to support sorting

BrandonTJS committed:
Merge pull request #263 from CS2103JAN2017-T09-B4/v0.5_Brandon

JUnit Fix to support sorting

loveandbejoyful committed:
CommandBox and StatusBarFooter tests

loveandbejoyful committed:
Merge branch 'V0.5' of https://github.com/CS2103JAN2017-T09-B4/main into V0.5

BrandonTJS committed:
Implemented Hotkeymanager JUnit

BrandonTJS committed:
Fixed an issue where recurring tasks are not subjected to sorting

BrandonTJS committed:
Implemented Undo complete for recurring tasks

loveandbejoyful committed:
Add result display tests

Crsytral committed:
Improve NotificationManager class code quality

BrandonTJS committed:
Fixed an issue where completed recurring tasks dissapears from calendar

Crsytral committed:
Update Logic Class Diagram

Save cmd access Storage class directly. Logic Manager have access to
Storage class as well.

loveandbejoyful committed:
Fix JUnit

loveandbejoyful committed:
Merge branch 'V0.5' of https://github.com/CS2103JAN2017-T09-B4/main into V0.5

BrandonTJS committed:
Fixed completed recurring display bug

loveandbejoyful committed:
Collate

BrandonTJS committed:
Some optimization

loveandbejoyful committed:
Fix JUnit result display test

BrandonTJS committed:
Fixed errornoues test case

BrandonTJS committed:
Merge pull request #264 from CS2103JAN2017-T09-B4/v0.5_Brandon

Multiple fixes + New implementations

loveandbejoyful committed:
Update collate

loveandbejoyful committed:
Merge branch 'V0.5' of https://github.com/CS2103JAN2017-T09-B4/main into V0.5

BrandonTJS committed:
Reduced delay that might have affected travis test

BrandonTJS committed:
Merge pull request #266 from CS2103JAN2017-T09-B4/v0.5_Brandon

Reduced delay that might have affected travis test

BrandonTJS committed:
Added support for recurring task into Task Builder

BrandonTJS committed:
Removed unused methods

BrandonTJS committed:
Testing a fix

BrandonTJS committed:
Changed test slightly

BrandonTJS committed:
Some Edit NLP test

BrandonTJS committed:
Testing a fix again

Crsytral committed:
Add Unit Test for Task

loveandbejoyful committed:
Add guitests

loveandbejoyful committed:
Merge branch 'V0.5' of https://github.com/CS2103JAN2017-T09-B4/main into V0.5

Crsytral committed:
Update ModelClassDiagram

Crsytral committed:
Update UIClassDiagram

Crsytral committed:
Update DG and UG

loveandbejoyful committed:
Pass some JUnit tests

loveandbejoyful committed:
Merge branch 'V0.5' of https://github.com/CS2103JAN2017-T09-B4/main into V0.5

loveandbejoyful committed:
Remove unused imports

Crsytral committed:
Update Author

BrandonTJS committed:
Updated Edit Command (Advance User)

BrandonTJS committed:
Updated find command

BrandonTJS committed:
Added additonal task details

BrandonTJS committed:
Cleaned up task class + Implemented recurring fields for edit descriptor

BrandonTJS committed:
Cleaned up RecurState

BrandonTJS committed:
Edit command support for recurring tasks

BrandonTJS committed:
Code cleanup and validation

BrandonTJS committed:
Fix Junit

BrandonTJS committed:
Merge pull request #267 from CS2103JAN2017-T09-B4/v0.5_Brandon

Removed unused method + testing a fix for hotkeymanager junit + Recurring task support for Task Builder

BrandonTJS committed:
Test fix

BrandonTJS committed:
Dev guide author tag

BrandonTJS committed:
Merge pull request #268 from CS2103JAN2017-T09-B4/v0.5_Brandon

Hotkeymanager new test

BrandonTJS committed:
Fixed JUnit

BrandonTJS committed:
Merge pull request #269 from CS2103JAN2017-T09-B4/v0.5_Brandon

Fixed JUnit

BrandonTJS committed:
Removed ArgumentTokenizer as it is not used

BrandonTJS committed:
New hotkey manager test

BrandonTJS committed:
Dev guide update

BrandonTJS committed:
2 new test class for logic parser

BrandonTJS committed:
Code cleanup

BrandonTJS committed:
Another HotkeyManager test

BrandonTJS committed:
code cleanup

ang-zy committed:
Create method in Model to update multiple tasks

ang-zy committed:
Update CompleteCommand to not refresh multiple times

ang-zy committed:
Merge branch 'Fix-Complete-Multi-Refresh' into V0.5

BrandonTJS committed:
Merge pull request #270 from CS2103JAN2017-T09-B4/v0.5_Brandon

Code cleanup + new junit tests + small dev guide update

ang-zy committed:
Merge branch 'Recurring-Task-Add-Cmd' into V0.5

ang-zy committed:
Merge pull request #272 from CS2103JAN2017-T09-B4/V0.5

Merge V0.5 into Recurring-Task-Add-Cmd

loveandbejoyful committed:
Convert HashSet to LinkedHashSet to maintain FindCommand keyword sequence

loveandbejoyful committed:
Merge branch 'V0.5' of https://github.com/CS2103JAN2017-T09-B4/main into V0.5

Crsytral committed:
Update AboutUs

ang-zy committed:
Merge remote-tracking branch 'refs/remotes/origin/V0.5' into Recurring-Task-Add-Cmd

ang-zy committed:
Refactor code

BrandonTJS committed:
Fixed Recurring task support for complete command

ang-zy committed:
Error fix

ang-zy committed:
Merge remote-tracking branch 'refs/remotes/origin/V0.5' into Recurring-Task-Add-Cmd

BrandonTJS committed:
Commented

BrandonTJS committed:
Merge branch 'V0.5' into v0.5_Brandon

BrandonTJS committed:
Merge pull request #273 from CS2103JAN2017-T09-B4/v0.5_Brandon

Fixed recurring support for complete command

loveandbejoyful committed:
Merge pull request #274 from CS2103JAN2017-T09-B4/V0.5

V0.5

Crsytral committed:
Fix Checkstyle Violations

loveandbejoyful committed:
Fix checkstyle

loveandbejoyful committed:
Merge branch 'master' of https://github.com/CS2103JAN2017-T09-B4/main

BrandonTJS committed:
Added contraints to task details

Crsytral committed:
Update NotificationManager code quality

Crsytral committed:
Fix collate links not working in AboutUs.md

BrandonTJS committed:
Added Hotkey toggling to UG

Crsytral committed:
Improve Code Quality

BrandonTJS committed:
Edits to edit command ug

BrandonTJS committed:
Find command UG

BrandonTJS committed:
Merge pull request #276 from CS2103JAN2017-T09-B4/v0.5_Brandon

UG Update

BrandonTJS committed:
Formatting

BrandonTJS committed:
Formatting

BrandonTJS committed:
UG update

BrandonTJS committed:
Merge pull request #278 from CS2103JAN2017-T09-B4/v0.5_Brandon

UG Update

Crsytral committed:
Update UG with System tray notification

BrandonTJS committed:
Added comments for methods

BrandonTJS committed:
Merge pull request #279 from CS2103JAN2017-T09-B4/v0.5_Brandon

Added comments for methods

Crsytral committed:
Fix Checkstyle Violation

Crsytral committed:
Fix Checkstyle Violation

BrandonTJS committed:
Updated About Us

BrandonTJS committed:
Update UG,DG

BrandonTJS committed:
Merge pull request #280 from CS2103JAN2017-T09-B4/v0.5_Brandon

Update docs

BrandonTJS committed:
README update

BrandonTJS committed:
Formatting

BrandonTJS committed:
Recurring task addition

BrandonTJS committed:
Merge pull request #281 from CS2103JAN2017-T09-B4/v0.5_Brandon

README UPDATE

BrandonTJS committed:
Docs Update

BrandonTJS committed:
Merge pull request #282 from CS2103JAN2017-T09-B4/v0.5_Brandon

Docs Update

Crsytral committed:
Add NotificationManager Unit Test

Crsytral committed:
Fix Checkstyle Violation

ang-zy committed:
Add support for Recurring tasks in AddCommand

ang-zy committed:
Update test cases

ang-zy committed:
Merge branch 'master' into Recurring-Task-Add-Cmd

ang-zy committed:
Merge pull request #283 from CS2103JAN2017-T09-B4/Recurring-Task-Add-Cmd

Add support for Recurring task in add cmd

BrandonTJS committed:
Checkstyle + Hacky Recurring JUnit

ang-zy committed:
Fix import errors

BrandonTJS committed:
Merge branch 'master' into v0.5_Brandon

BrandonTJS committed:
Merge pull request #284 from CS2103JAN2017-T09-B4/v0.5_Brandon

Checkstyle + Hacky Recurring JUnit

BrandonTJS committed:
Checkstyle + Some Recurring Junit

ang-zy committed:
Fix checkstyle violations

loveandbejoyful committed:
Rename JUnit

loveandbejoyful committed:
Merge branch 'master' of https://github.com/CS2103JAN2017-T09-B4/main

BrandonTJS committed:
Merge pull request #285 from CS2103JAN2017-T09-B4/v0.5_Brandon

Checkstyle + Some Recurring Junit

loveandbejoyful committed:
Merge branch 'master' of https://github.com/CS2103JAN2017-T09-B4/main

Crsytral committed:
Fix Checkstyle Violation

Crsytral committed:
Update NotificationManager Unit Test

loveandbejoyful committed:
Checkstyle

Crsytral committed:
Update NotificationManager Unit Test

loveandbejoyful committed:
Resolve merge conflicts

loveandbejoyful committed:
Merge branch 'master' of https://github.com/CS2103JAN2017-T09-B4/main

ang-zy committed:
Update Developer Guide with Undo implementation

ang-zy committed:
Fix merge conflicts

# Conflicts:
#	src/main/java/seedu/tache/logic/parser/AddCommandParser.java
#	src/main/java/seedu/tache/logic/parser/ParserUtil.java
#	src/test/java/guitests/AddCommandTest.java

ang-zy committed:
Refactor UndoHistory into its own class

loveandbejoyful committed:
Add back recurring tasks

loveandbejoyful committed:
Checkstyle

ang-zy committed:
Update Test case for UndoCommand

Fixes #121

Crsytral committed:
Update Unit Test for SaveAndLoadCommandTest

Crsytral committed:
Update Unit Test for NotificationManagerTest

BrandonTJS committed:
Renamed

BrandonTJS committed:
Merge pull request #286 from CS2103JAN2017-T09-B4/v0.5_Brandon

Renamed

BrandonTJS committed:
Missed out 1

BrandonTJS committed:
Merge pull request #287 from CS2103JAN2017-T09-B4/v0.5_Brandon

Missed out 1

Crsytral committed:
Update comments

Crsytral committed:
Update Comments

ang-zy committed:
Refactor DateTime parser to match SLAP

ang-zy committed:
Refactor AddCommandParser to better meet SLAP

ang-zy committed:
Refactor AddCommandParser to better meet SLAP

ang-zy committed:
Remove deprecated structured command parsing for AddCommand

ang-zy committed:
Remove deprecated structured command parsing for AddCommand

BrandonTJS committed:
Bug fix

BrandonTJS committed:
Merge pull request #288 from CS2103JAN2017-T09-B4/v0.5_Brandon

Bug fix

ang-zy committed:
Update Documentation

ang-zy committed:
Add Test case for AddCommandParser

Crsytral committed:
Update AboutUs

ang-zy committed:
Update Documentation for UndoHistory

ang-zy committed:
Add author

ang-zy committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Documentation+Test_Zhi-Yuan

# Conflicts:
#	src/main/java/seedu/tache/logic/parser/AddCommandParser.java

ang-zy committed:
Merge pull request #289 from CS2103JAN2017-T09-B4/Documentation+Test_Zhi-Yuan

Update Documentation + Test Case

loveandbejoyful committed:
Edit language

ang-zy committed:
Update SD to reflect refactored code

ang-zy committed:
Remove red lines

ang-zy committed:
Update weblink on user guide

ang-zy committed:
Update Add Command usage instructions for recurring tasks

ang-zy committed:
Update User Guide for adding recurring tasks

BrandonTJS committed:
Stop sample data from loading

BrandonTJS committed:
Added Manual Testing docs

BrandonTJS committed:
Merge pull request #290 from CS2103JAN2017-T09-B4/v0.5_Brandon

Stop sample data + Manual testing stuff

loveandbejoyful committed:
Edit UI screenshots

loveandbejoyful committed:
Merge branch 'master' of https://github.com/CS2103JAN2017-T09-B4/main

ang-zy committed:
Update AboutUs

BrandonTJS committed:
Collate update

ang-zy committed:
Fix Checkstyle Violation

BrandonTJS committed:
Revert "Collate update"

This reverts commit 6c1f43facde56e27736f7b7161c6534a4cca356f.

BrandonTJS committed:
Fix taskmanger not loading properly

BrandonTJS committed:
Merge pull request #291 from CS2103JAN2017-T09-B4/v0.5_Brandon

Fix taskmanager load problem

loveandbejoyful committed:
Collate

ang-zy committed:
Fixed formatting issue

loveandbejoyful committed:
Collate again

loveandbejoyful committed:
Merge branch 'master' of https://github.com/CS2103JAN2017-T09-B4/main

BrandonTJS committed:
Fix Junit

BrandonTJS committed:
Merge pull request #292 from CS2103JAN2017-T09-B4/master_final

Fix Junit

loveandbejoyful committed:
Some minor errors
